src/Extension.ml

```diff
@@ -94,6 +96,11 @@ module Instance = struct
         "ocamllsp in this toolchain is out of date, functionality will not be \
          available in mli sources. Please update to a recent version and \
          restart the server.";
+    if not (OcamlLsp.handleSwitchImplIntf ocamlLsp) then
+      message `Warn
+        "ocamllsp in this toolchain is out of date, switching between \
+         implementation and interface files will use a less accurate fallback \
+         mechanism. Please update to a recent version and restart the server.";
```

Comment:
We should combine these the two messages above into:

```
ocamllsp in this toolchain is out of date. The following features might be unavailable or degraded in functionality:
- switching between implementation and interface files
- functionality in mli sources
```

Reply:
The VSCode message boxes strip newlines, so I replaced them with commas.
> ocamllsp in this toolchain is out of date. The following features might be unavailable or degraded in functionality: switching between implementation and interface files, functionality in mli sources